### PR TITLE
Add angle_difference.

### PR DESCRIPTION
**Pull request by tkoolen**
Useful utility for working with angles.

### CODE REVIEW

File: src/util.jl [hunk 1]
```diff
@@ -19,3 +19,12 @@ function perpendicular_vector(vec::SVector{3})
     @inbounds perpind2 = vec[ind1]
     perp = @SVector [ifelse(i == ind1, perpind1, ifelse(i == ind2, perpind2, zero(T))) for i = 1 : 3]
 end
+
+"""
+    angle_difference(a, b)
+
+Compute the difference between two angles in such a way that:
+* for angles `a` and `b` (in radians), and `c = angle_difference(a, b)`, `cos(a) ≈ cos(b + c)` and `sin(a) ≈ sin(b + c)`;
+* the result is in the interval [-π, π).
```
Comment (c42f) [inline src/util.jl]: This is a useful utility (It's like deg2rad, I think I've written it several times before in various code bases :-) ).  I found this documentation a little confusing though.  How about something like:

"Compute the angle difference `a-b` wrapped into the interval [-π, π) by subtracting or adding a multiple of 2π.  This is the smallest number `c` such that `a` and `b+c` represent the same point on the circle, and likewise `b` and `a-c`."

Comment (c42f) [reply src/util.jl]: Hmm.  Smallest in the sense of `abs(c)` of course... could no doubt still be confusing.

Comment (tkoolen) [reply src/util.jl]: Done.

File: test/util_tests.jl [hunk 2]
```diff
@@ -7,4 +7,16 @@
...
         end
     end
+
+    @testset "Angle difference" begin
+        for i = 1 : 100
+            a = rand(-5π : 5π)
+            b = rand(-5π : 5π)
+            c = angle_difference(a, b)
+            @test isapprox(cos(b + c), cos(a); atol = 1e-12)
+            @test isapprox(sin(b + c), sin(a); atol = 1e-12)
```
Comment (c42f) [inline test/util_tests.jl]: The following syntax is also available for this, by the way:

```
@test cos(b+c)≈cos(a)  atol=1e-12
```

**Comment by c42f** [comment]
@andyferris any thoughts about having this as part of the public API? My thought is that it's pretty useful, if not quite strictly related to rotations per se.

@tkoolen The CI failures seem spurious and confusing on 0.6, I restarted Travis in the hope that it will go through.  It's all broken with 0.7 of course, probably due to Base breaking StaticArrays :-/

File: src/util.jl (hunk 1 above)
Comment (c42f) [reply src/util.jl]: Ah nice, you fixed up my awkward wording regarding the absolute value.

**Comment by tkoolen** [comment]
The CI failure, which is due to the HDF5 issue I was trying to work around with https://github.com/FugroRoames/Rotations.jl/pull/46. I remember rebasing onto master since that PR was merged, but I might have forgotten to pull master first. Anyway, rebased now; hopefully that fixes it.

**Comment by andyferris** [comment]
@tkoolen did you see that base has `rem2pi(a - b, RoundNearest)`?

When I saw this PR I thought it should be called something like `mod2pi` or something (well, I guess it does the subtraction for you...) but it seems like `Base` beat me to it. I'm not strictly *against* this but there's always some overhead with any new API so it's worth designing well.

**Pull request closed without merging**